app/lib/animation.ts

```diff
@@ -0,0 +1,123 @@
+import { clamp } from "./math";
+import { TimeDelta } from "./time";
+
+export class AnimationLoop {
```

Comment:
maybe we can put this in an animation / chart / visualization specific package? could be the same one as the flame chart

Reply:
I moved it to its own target, along with the other files -- if we start adding more stuff to this class then I'll split into more files / a separate dir, does that sound good?